demo/project_demo09-S3/run.py

```diff
@@ -0,0 +1,128 @@
+# Read all objects with the prefix "foo" from an S3 bucket. The S3 bucket is pre-loaded with two objects,
+# foo_1.csv and foo_2.csv, that have values (1, 2, 3) and (4, 5, 6).
+#
+# It expects two environment varaibles CI_S3_AWS_ACCESS_KEY and CI_S3_AWS_SECRET for the test IAM user permissions.
+#
+# To run the demo, start Feldera:
+# > docker compose -f deploy/docker-compose.yml -f deploy/docker-compose-dev.yml --profile s3-demo \
+#       up db pipeline-manager --build --renew-anon-volumes --force-recreate
+#
+# Run this script:
+# > python3 run.py --api-url=http://localhost:8080 --start
+import base64
+import os
+import time
+import datetime
+import requests
+import argparse
+from plumbum.cmd import rpk
+import psycopg
+import json
+import random
+
+# File locations
+SCRIPT_DIR = os.path.join(os.path.dirname(__file__))
+PROGRAM_SQL = os.path.join(SCRIPT_DIR, "program.sql")
+
+DATABASE_NAME = "jdbc_test_db"
+PIPELINE_NAME = "demo-s3"
+FELDERA_CONNECTOR_NAME = "s3-input"
+TABLE_NAME = "test_table"
+VIEW_NAME = "test_view"
+PRIMARY_KEY_COLUMN = "id"
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--api-url", required=True, help="Feldera API URL (e.g., http://localhost:8080 )")
+    parser.add_argument('--start', action='store_true', default=False, help="Start the Feldera pipeline")
+    args = parser.parse_args()
+    prepare_feldera_pipeline(args.api_url, args.start)
+
+def prepare_feldera_pipeline(api_url, start_pipeline):
+    s3_access_key = os.getenv("CI_S3_AWS_ACCESS_KEY")
+    s3_secret_key = os.getenv("CI_S3_AWS_SECRET")
+    assert s3_access_key is not None and s3_secret_key is not None
+
+    # Create program
+    program_name = "demo-s3-input"
+    program_sql = open(PROGRAM_SQL).read()
+    response = requests.put(f"{api_url}/v0/programs/{program_name}", json={
+        "description": "S3 input connector demo",
+        "code": program_sql
+    })
+    response.raise_for_status()
+    program_version = response.json()["version"]
+
+    # Compile program
+    print(f"Compiling program {program_name} (version: {program_version})...")
```

Comment:
I asked elsewhere too, shouldn't these be made into a library of Python utilities?
Would make it much easier to change things, e.g., urls.

Reply:
Making this a library without distributing it means it's harder for users to run our demos. Making the demos self-contained like this makes it clear how the APIs work.

Reply:
Why not distribute the library?